firebase-plugin/index.ts

```diff
@@ -0,0 +1,9 @@
+import firebase from "firebase";
+import { IFirebasePluginConfig } from "./interfaces/firebase-plugin-config";
+import { writeToDB } from "./write-to-db";
+
+export default (cfg: IFirebasePluginConfig) => (report: PerryReport) => {
+  const basePath = cfg.basePath ? cfg.basePath : "perry_reports";
```

Comment:
instead of doing `cfg.someThing ? cfg.someThing : "default_value"`, would be better to have a `const defaultConfig` which defines those, such as:

```ts
import firebase from "firebase";
import { IFirebasePluginConfig } from "./interfaces/firebase-plugin-config";
import { writeToDB } from "./write-to-db";

const defaultConfig: IFirebasePluginConfig = {
  basePath: "perry_reports",
  transform: (report: any) => report;
};

export default (cfg: IFirebasePluginConfig) => (report: PerryReport) => {
  const actualConfig: IFirebasePluginConfig = {
    ...cfg,
    ...defaultConfigValues
  };

  const basePath = actualConfig.basePath;
  const ref = actualConfig.firebaseApp.database().ref(`${basePath}/${report.id}`);
  writeToDB(ref, actualConfig.transform(report));
}
```

Reply:
I think it will help better understanding and extending this code, since it leaves this with less cyclomatic complexity or conditional flows =)